internal/extension/smartagentextension/factory.go

```diff
@@ -64,6 +66,12 @@ func createDefaultConfig() config.Extension {
 	}
 	cfg.BundleDir = bundleDir
 	cfg.Collectd.BundleDir = bundleDir
+
+	dir := strings.TrimSpace(os.Getenv(collectdConfigDirEnvVarName))
+	if dir != "" && strings.TrimSpace(cfg.Collectd.ConfigDir) == "" {
```

Comment:
A few things going on here:

`cfg.Collectd.ConfigDir` is the agent's default value so this empty string value should never happen: https://github.com/signalfx/signalfx-agent/blob/master/pkg/core/config/config.go#L387

The custom unmarshaller makes the default config not actually the default and needs to be pulled from directly like is happening w/ the bundleDir here: https://github.com/signalfx/splunk-otel-collector/blob/main/internal/extension/smartagentextension/config.go#L56

I'm not 100% we want to use that env var value here since it's just used by the connector installation path (I see that the readme is vague though).  What if we were to use `Filepath.Join(cfg.BundleDir, "run", "collectd")` in the custom unmarshaller and leave this code unchanged?

Reply:
You could also take care of this in a separate PR since it's not required for bundling in docker afaict and just rely on your added `ENV SPLUNK_COLLECTD_DIR=$SIGNALFX_BUNDLE_DIR/run/collectd` line, unless that's not taking effect.

Reply:
I deleted this code and moved the logic to configure collectd config dir to smartagentextension/config.go.